records/generator.go

```diff
@@ -120,28 +128,31 @@ func (rg *RecordGenerator) loadFromMaster(ip string, port string) (sj state.Stat
 	req, err := http.NewRequest("GET", u.String(), nil)
 	if err != nil {
 		logging.Error.Println(err)
+		return state.State{}, err
 	}
 
 	req.Header.Set("Content-Type", "application/json")
 
-	client := &http.Client{}
-	resp, err := client.Do(req)
+	resp, err := rg.httpClient.Do(req)
 	if err != nil {
 		logging.Error.Println(err)
+		return state.State{}, err
 	}
 
+	defer util.IgnoreError(resp.Body.Close)
```

Comment:
This is mostly a matter of dealing with the linters. Related: https://github.com/mesosphere/mesos-dns/issues/223 -- I don't think it's a good idea to ignore those errors completely. I think that being explicit about ignoring errors is valuable, and as laid out in the issue, we'd want to handle R/W objects differently.